crates/chia-consensus/src/run_block_generator.rs

```diff
@@ -154,6 +157,13 @@ fn extract_n<const N: usize>(
     Ok(ret)
 }
 
+// this function checks if the generator start with a quote
+// this is required after the SIMPLIFY_GENERATOR fork is active
+#[inline]
+pub fn check_generator_quote(program: &[u8], flags: u32) -> bool {
```

Comment:
if you make this function return `Result<(), ValidationErr>`, calling it will be much simpler, just:

```
check_generator_quote(program, flags)?;
```